Task2/usersAPI.ts

```diff
@@ -0,0 +1,85 @@
+import express from 'express';
+import uuidv4 from 'uuid/v4';
+
+import { usersCollection } from './usersCollection';
+import validateUser from './checkValidation';
+
+const app: express.Application = express();
+const router = express.Router();
+const port = process.env.PORT || 3000;
+
+app.use(express.json());
+app.use('/users', router);
+
+function getAutoSuggestUsers (loginSubstringin:string, limit: string):any {
+	if (!loginSubstringin || loginSubstringin.length === 0) return usersCollection;
+	loginSubstringin = loginSubstringin.toLowerCase();
+	const numLimit: number = limit ? +limit : usersCollection.length;
+
+	const filteredAndSortedUsersCollection = usersCollection.filter(user => {
+		return user.login.startsWith(loginSubstringin);
+	}).sort((userA, userB) => {
+		return userA.login < userB.login ? -1 : 1;
+	});
+	const limitCollection = Math.min(
```

Comment:
I would use `collectionLimit` name instead of  `limitCollection`, because this is common practice to use verbs for functions and nouns for variables.